packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -352,7 +352,15 @@ class FlutterCommandRunner extends CommandRunner<Null> {
         Uri rootUri = flutterUri.resolve('../../..');
         String flutterPath = path.normalize(fs.file(rootUri).absolute.path);
 
-        if (!_compareResolvedPaths(flutterPath, Cache.flutterRoot)) {
+        if (!fs.isDirectorySync(rootUri.path)){
+          // Remove the .packages file that now points to an obsolete SDK location.
+          fs.file(kPackagesFileName).deleteSync();
```

Comment:
re: this and comment 274691835, you're right, it does run packages get automatically. I missed it before. 

So missing sdk and the multiple sdks check is basically the same first principle: the place previously referenced in pubspec is wrong. Merged to keep things simple. Also removed the .packages delete. It's not necessary.